### ISSUE TITLE
Authentication examples

### ISSUE THREAD

**Issue by planetmarshall** [zeromq/cppzmq#409]
It would be useful to have some examples showing usage of the CurveZMQ protocol in cppzmq - similar to `stonehouse` etc that exist in CZMQ, PyZMQ and others.

If nothing already exists, I could do this as I am currently working in that area.

Thanks,
Andrew.

**Comment by brettviren** [comment]
+1 encouragement from me.  Having some worked cppzmq "houses" examples would be very useful.

**Comment by gummif** [comment]
I agree. It might also be useful to implement a generic zap server in cppzmq, but I have no experience with that, so cant tell how complicted it would be.

**Comment by sigiesec** [comment]
We definitely lack good examples, so any contribution would be very welcome!

While I implemented a ZAP server in the past and it was not too complex, doing that in a generic way would be much more complex, as you would need to provide some API for integrating with authentication providers.

**Comment by kassem-tohme** [comment]
Can anyone provide an example on PLAIN authentication with cppzmq? Me and maybe others would appreciate it.

**Comment by Teebonne** [comment]
You might want to look at [this](https://github.com/brettviren/cppzmq-houses) and [this](http://www.evilpaul.org/wp/2017/05/02/authentication-encryption-zeromq/).
